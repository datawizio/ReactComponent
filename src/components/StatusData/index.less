.StatusData__container {
  color: rgba(255, 255, 255, 0.65);
  font-weight: bold;
  position: fixed;
  bottom: 0;
  background: #09011c;
  padding: 22px 0 14px 22px;
  font-size: 12px;
  width: 232px;
  display: block;

  p {
    margin-bottom: 0;
  }

  p:last-child {
    font-style: italic;
  }
}

.ant-layout-sider-collapsed .StatusData__container {
  display: none;
}

.rtl {
  .StatusData__container {
    padding: 22px 22px 14px 0;
  }
}