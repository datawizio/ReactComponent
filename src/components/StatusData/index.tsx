import React, { useContext } from "react";
import ConfigContext from "../ConfigProvider/context";
import { formatDateTime } from "../../utils/date";
import "./index.less";

interface StatusDataProps {
  inProcess: boolean;
  lastUpdateDate: string | null;
}

const StatusData: React.FC<StatusDataProps> = ({
  inProcess,
  lastUpdateDate
}) => {
  const { translate } = useContext(ConfigContext);

  if (inProcess) {
    return (
      <div className="StatusData__container">
        <p>{translate("DATA_IN_PROGRESS")}</p>
      </div>
    );
  }

  if (lastUpdateDate) {
    const formattedDateTime = formatDateTime(lastUpdateDate, translate);
    return (
      <div className="StatusData__container">
        <p>{formattedDateTime}</p>
        <p>{translate("LAST_UPDATE_DATA")}</p>
      </div>
    );
  }

  return null;
};

export default StatusData;
