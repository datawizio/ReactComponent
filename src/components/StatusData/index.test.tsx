import "jsdom-global/register";
import React from "react";
import { shallow } from "enzyme";

import StatusData from "./index";

const mockProps = {
  inProcess: false,
  lastUpdateDate: "2021-06-01 04:02:37"
};

const setUp = (props?) => shallow(<StatusData {...props} />);

describe(" StatusData component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("StatusData rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("StatusData in process", () => {
    const wrapper = setUp({ ...mockProps, inProcess: true });
    expect(wrapper.html()).toContain("DATA_IN_PROGRESS");
  });
});
