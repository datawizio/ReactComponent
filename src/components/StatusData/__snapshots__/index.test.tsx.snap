// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` StatusData component StatusData rendered correctly 1`] = `
<div
  className="StatusData__container"
>
  <p>
    01.06.2021 04:02
  </p>
  <p>
    LAST_UPDATE_DATA
  </p>
</div>
`;
